runtime/src/main/java/io/quarkiverse/ironjacamar/runtime/endpoint/TransactionAwareMessageEndpoint.java

```diff
@@ -79,4 +84,28 @@ public void afterDelivery() {
     public void release() {
         // Do nothing
     }
+
+    /**
+     * Get the current transaction status
+     */
+    int getStatus() {
+        try {
+            return UserTransaction.userTransaction().getStatus();
```

Comment:
Haven't checked the docs, but is it possible that `userTransaction()` returns `null` in some cases?

Reply:
Good catch!  I think if it is `null`, there could be something wrong with the transaction context (e.g. it is finished early?). @mmusgrov WDYT?

Reply:
Sorry @zhfeng and @gastaldi I've just seen this. According to the spec it cannot be null. If there is no transaction context then the [status() call](https://jakarta.ee/specifications/platform/9/apidocs/jakarta/transaction/usertransaction) returns Status.NoTransaction or SystemException if the transaction manager encounters an unexpected error condition ie the code looks fine.